### ISSUE TITLE
False negative attribute resolution of function's argument

### ISSUE THREAD

**Issue by stanislavlevin** [PyCQA/pylint#4242]
### Steps to reproduce

Given a file `a.py`:
```python
def test(val):
    val.somethingwrong()

test(1)
```

### Current behavior
Result of `pylint a.py`:
```
------------------------------------
Your code has been rated at 10.00/10
```

### Expected behavior
Pylint should report an error.

### pylint --version output

Result of `pylint --version` output:
```
pylint 2.7.2
astroid 2.5.1
Python 3.9.2 (default, Feb 26 2021, 15:33:46) 
[GCC 10.2.1 20201125 (ALT Sisyphus 10.2.1-alt2)]
```

**Comment by hippo91** [comment]
@stanislavlevin thanks for your report. I don't think it is a pylint bug. The only way to detect an error would be to detect that test parameter is 1 and so doesn't have `somethingwrong` method. However doing this would necessitate that `astroid`, the `pylint`'s inference engine has control-flow analysis ability. This is not yet the case and it is really a huge work.

**Comment by Pierre-Sassoulas** [comment]
`mypy` does not find this either and typing is its specialty. I think it's safe to say you need to at least add typing if you want proper reporting.

Ie:
```python

class MyClass:
    def somethingwrong(self):
        pass

def test(val:MyClass):
    val.somethingwrong()

test(1)
```
mypy:
```
a.py:9: error: Argument 1 to "test" has incompatible type "int"; expected "MyClass"

```

**Comment by stanislavlevin** [comment]
I've already realized that it is mypy task and without annotations it is unsolvable at all. So if there are no plans to add support for type hints in Pylint then please close this issue.

**Comment by Pierre-Sassoulas** [comment]
Closing as a duplicate of https://github.com/PyCQA/pylint/issues/4813
